lib/lattice_observer/observed/lattice.ex

```diff
@@ -453,19 +453,42 @@ defmodule LatticeObserver.Observed.Lattice do
     l
   end
 
+  # pk is an optional filter by public key
   @spec running_instances(
           LatticeObserver.Observed.Lattice.t(),
           nil | String.t(),
           String.t()
         ) :: [%{id: String.t(), instance_id: String.t(), host_id: String.t()}]
   def running_instances(%Lattice{} = l, pk, spec_id) when is_binary(pk) do
     if String.starts_with?(pk, "M") do
-      actors_in_appspec(l, spec_id)
+      actors = actors_in_appspec(l, spec_id)
+
+      actors =
+        if pk != nil do
+          actors
+          |> Enum.filter(fn %{actor_id: pub_key, instance_id: _, host_id: _} -> pub_key == pk end)
```

Comment:
I think you can leave the other fields entirely out of the map. It doesn't require you to define them all, so you could do:
```
fn %{actor_id: pub_key} -> ...
```
and it should still match if there are other fields in the map.